### PR TITLE
Enhancements and changes

### PR DESCRIPTION
**Pull request by sebasbaumh**
Thanks for your effort in creating a thin wrapper over OpenLayers 3.
I enhanced it to be able to use it in a project, which formerly used OpenLayers 2.
In detail I added missing control and geometry classes and functions on the existing ones.

The main classes to get started are your OLFactory class and OLUtil, which provides some utility functions for event handlers (which I made similar to OpenLayers 2).

### CODE REVIEW

**Comment by TDesjardins** [comment]
Hello Sebastian,

thanks for your efforts. There are some nice improvements. By the way I have some remarks. Your code have no consequent codestyle, so I think we have to define some basic rules and check these ones with checkstyle. Furthermore this is a very big pull request, so it is hard to review all the changes. I currently work on an update for gwt-ol3 for the final jsinterop features GWT 2.8. It would be nice if you can provide some contact information. 

Regards

Tino

**Comment by sebasbaumh** [comment]
Hi Tino,

First of all thank you for creating this good base for an OpenLayers 3 wrapper.
I also had a look on [Vaadin's OpenLayers 3 wrapper](https://github.com/VOL3/v-ol3), but that did use the old JSNI interfaces and still not fully matched my requirements.
Therefore I got started on your implementation as using the new jsinterop features looked pretty promising.

I finally used the default Java codestyle in eclipse. But I would also appreciate a fixed definition to be able to apply it to all files.

Sorry for the big pull request - I know, I reworked a lot of things.
Basically I worked my way through most of the classes trying to resemble the usage of OpenLayers 2 in one of our company's projects.
After that I cleaned up the other ones to some extent.

You can reach me via sebastian.baumhekel@gmail.com.

Best regards and have a nice weekend,

Sebastian
